java/java-ipfs.md

```diff
@@ -0,0 +1,489 @@
+# IPFS and Java
+
+tags: java, ipfs, ipns
+
+====
+
+In this article, we will learn how to interract with **[IPFS](https://ipfs.io/) (InterPlanetary File System)** in Java using the official [**java-ipfs-http-client library**](https://github.com/ipfs/java-ipfs-http-client). This library connects to an IPFS node and wraps most of the operations offered by the [HTTP API](https://docs.ipfs.io/reference/api/http/).
+
+The following diagram describes Java program connected to an IPFS node via the **java-ipfs-http-client** library to the API Server. 
+
+![](https://imgur.com/RRB6chj.png)
+
+- API server (default port: 5001): Full API
+- Gateway server (default port: 8080): Read Only API (access to data only)
+- P2P (default port: 4001): Peer-to-peer interface
+
+
+
+<br />
+
+## Prerequisite
+
+To run this tutorial, we must have the following installed:
+
+- Java programming language (> 8)
+```shell
+$ java -version
+java version "1.8.0_201"
+```
+
+- A package and dependency manager, for example [Maven](https://maven.apache.org) or [Gradle](https://gradle.org)
+
+- An IDE (Integrated development environment), for this tutorial, we use Eclipse
+
+- A running IPFS node (> 0.4.x)
+*Follow [the following article](https://kauri.io/article/b01b9b7bebcd4ebf80edf021bdd0e232/v2/installing-ipfs) to learn how to install an IPFS node (go-ipfs)*
+```shell
```

Comment:
As there's a few different ways to start a node, and you also need to run `ipfs init` first anyway, I'd say, just remove this code snippet and leave the link instead for more information.